jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpDestination.java

```diff
@@ -388,8 +388,7 @@ private boolean process(Connection connection)
             LOG.debug("Processing exchange {} on {} of {}", exchange, connection, this);
         if (exchange == null)
         {
-            if (!connectionPool.release(connection))
-                connection.close();
+            releaseConnection(connection);
```

Comment:
The else clause below is not necessary and makes this code a little harder to read.  Without it, it would be 4 if statements, each returning.

But this is unrelated to this PR

Reply:
The current structure of the code is:
```java
    if (A) 
    {
        // do A
        return;
    }
    else
    {
          if (B)
          {
               // do B
               return;
          }

          if (C)
          {
               // do C
               return;
          }

          // do D
    }
```
This is needlessly convoluted and would be much clearer as:
```java
    if (A) 
    {
        // do A
        return;
    }

    if (B)
    {
         // do B
         return;
    }

    if (C)
    {
         // do C
         return;
    }

    // do D
```